### ISSUE TITLE
Accessing a database in the API has inconsistent API

### ISSUE THREAD

**Issue by johnpapa** [Azure/azure-cosmos-js#122]
Sometimes we need just the id of the database, and sometimes we need an object with an id property with the name of the database.

The error, when calling the following code is `error: invalid string input`

```
  const databaseDefName = 'my db';
  const { database } = await client.databases.createIfNotExists(databaseDefName);
```

So instead we need to do this

```
  const databaseDefName = 'my db';
  const { database } = await client.databases.createIfNotExists({
    id: databaseDefName
  });
```

But for the other APIs, such as this, we can simply do ...

```
  const databaseDefName = 'my db';
  const db = await client.database(databaseDefName);
```

It would be nice to have a consistent API and also clearer error messages on these

cc @christopheranderson

**Comment by christopheranderson** [comment]
I was thinking about supporting this for databases since you do mostly just need a string. In the case of containers, I do want to encourage folks to think about the options they should set.

Also, I plan on supporting taking in a property bag with "id" on it for things that need an id as string passed in. See https://github.com/Azure/azure-cosmos-js/issues/42

**Comment by southpolesteve** [comment]
Closed in favor of #42
